exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1326 @@
+package bybit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/config"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/account"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/deposit"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/orderbook"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/protocol"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/stream"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/ticker"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/trade"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+// GetDefaultConfig returns a default exchange config
+func (by *Bybit) GetDefaultConfig() (*config.Exchange, error) {
+	by.SetDefaults()
+	exchCfg := new(config.Exchange)
+	exchCfg.Name = by.Name
+	exchCfg.HTTPTimeout = exchange.DefaultHTTPTimeout
+	exchCfg.BaseCurrencies = by.BaseCurrencies
+
+	err := by.SetupDefaults(exchCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	if by.Features.Supports.RESTCapabilities.AutoPairUpdates {
+		err := by.UpdateTradablePairs(context.TODO(), true)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return exchCfg, nil
+}
+
+// SetDefaults sets the basic defaults for Bybit
+func (by *Bybit) SetDefaults() {
+	by.Name = "Bybit"
+	by.Enabled = true
+	by.Verbose = true
+	by.API.CredentialsValidator.RequiresKey = true
+	by.API.CredentialsValidator.RequiresSecret = true
+
+	requestFmt := &currency.PairFormat{Uppercase: true}
+
+	configFmt := &currency.PairFormat{Uppercase: true}
```

Comment:
Config storage of pairs will require a delimiter `Bybit ticker sync complete DOG-EUSDT [7/20].` When it is saved to the config it then gets broken down via our method unmarshal json and calls this: https://github.com/thrasher-corp/gocryptotrader/blob/105548511e15c9419c45e0c7e31bf1131cbe2a5a/currency/pair.go#L98

So it should be in memory saved as DOGE - USDT instead.

Reply:
Okay. So we need this right? @shazbert 

`configFmt := &currency.PairFormat{Delimiter: currency.DashDelimiter, Uppercase: true}`

Reply:
@Jdpurohit Yeah, that's exactly what we need. 👍

Reply:
Hello @shazbert 
Can you please help me better understand what is the use of `requestFmt` and `configFmt` ?

I am asking this because after making the above changes and this change in `FetchTradablePairs`

`pairs = append(pairs, allPairs[x].BaseCurrency+currency.DashDelimiter+allPairs[x].QuoteCurrency)`

I am facing an issue when running GCT using `testdata/configtest.json`. As per my understanding, Bybit expects symbols like for example `BTCUSDT` and we are passing `BTC-USDT` 

Here is a log of the error.
`Failed to get REST orderbook for Bybit DOT-USDT spot. Error: Bybit unsuccessful HTTP status code: 400 raw response: {"ret_code":-100011,"ret_msg":"Not supported symbols","result":null,"token":null}`

GCT binary used for my testing was built on this commit. 
https://github.com/Jdpurohit/gocryptotrader/commit/2fb502d062ff4b93695146e5d72ca7205aa7b978

Reply:
```go
formattedPair, err := by.FormatExchangeCurrency(p, assetType)
if err != nil {
	return nil, err
}
```
The above needs to be included in Bybit's `UpdateOrderbook` method function. When you call method `FormatExchangeCurrency` it will then format it to the request style format `requestFmt ` for that asset type `SPOT` which is uppercase and does not include a delimiter. 

Unfortunately, we cannot store pairs without a delimiter due to the limitations of saving it to config as a string and then unmarshaling from there (and there is no standard anymore with regards to currency codes). Originally it was used to match config values to request value for performance reasons but due continuous updates, this is of no concern anymore. 

Added an item here for a currency update list: https://trello.com/c/54eizOIo/143-currency-package-upgrades